### DISCUSSION TITLE
How can I start the macOS executable?

### DISCUSSION THREAD

**Discussion by PolDescomm** [KevinBytesTheDust/CVAmp#159] [Q&A]
Hello, How can try the bot on macOS? I using OSx Ventura 13.4.1

**Comment by jlplenio** [comment]
You will have to start the tool from the Terminal. I will build and .app to start with a double click in the future.

**Comment by PolDescomm** [comment]
Can u do a mini tutorial for this please? ThanksEl 23 jun 2023, a las 16:20, jlplenio ***@***.***> escribió:﻿
You will have to start the tool from the Terminal. I will build and .app to start with a double click in the future.

**Comment by jlplenio** [comment]
**1. Open Terminal**

Open the Terminal application through Finder -> Applications -> Utilities -> Terminal, or use Spotlight (Command key + Spacebar) and type in "Terminal".

**2. Navigate to the Directory of Your .zip File**

Use the `cd` (change directory) command to navigate to the folder where your .zip file is. For example, if your .zip file is in your Downloads folder, you'd type:

    cd ~/Downloads

**3. Extract the .zip File**

Extract the .zip file using the `unzip` command. Assume your file is named `CTVBot0.6.0_20230622_macOS_experimental.zip`. Use the following command to extract it:

    unzip CTVBot0.6.0_20230622_macOS_experimental.zip

This will extract all the files into a directory with the same name (without the .zip extension).

**4. Navigate to the Extracted Directory**

After extraction, navigate into the directory:

    cd CTVBot0.6.0_20230622_macOS_experimental

**5. Change File Permissions**

Next, change the permissions of the .bin file to make it executable. The .bin file in this case is `CTVBot0.6.0_20230622_macOS_experimental.bin`. Use the following command to make it executable:

    chmod +x CTVBot0.6.0_20230622_macOS_experimental.bin

**6. Run the .bin File**

Now that the file is executable, you can run it using the `./` prefix, like so:

    ./CTVBot0.6.0_20230622_macOS_experimental.bin

That's it! The .bin file should now execute and run its program. 

Thanks @ChatGPT

**Comment by PolDescomm** [comment]
Hello Again, I receive this error message



./CTVBot0.6.0_20230622_macOS_experimental.bin

./CTVBot0.6.0_20230622_macOS_experimental.bin

Available proxies 0

Available window locations 2

Traceback (most recent call last):

  File "main_gui.py", line 24, in <module>

  File "ctvbot/gui.py", line 62, in __init__

  File "tkinter/__init__.py", line 2270, in __init__

_tkinter.TclError: Can't find a usable init.tcl in the following
directories:

    /usr/local/Cellar/tcl-tk/8.6.13_3/lib/tcl8.6
/Users/****/Downloads/lib/tcl8.6 /Users/poldescomm/lib/tcl8.6
/Users/****/Downloads/library /Users/****/library
/Users/****/tcl8.6.13/library /Users/tcl8.6.13/library




This probably means that Tcl wasn't installed properly.


[10752] Failed to execute script 'main_gui' due to unhandled exception!

Gui shutting down 2023-06-24 03:26:06.023696

**Comment by jlplenio** [comment]
That does not look like an easy fix. I will put in on my todo list at the bottom , sorry :/
